artifact/gem5art/artifact/_artifactdb.py

```diff
@@ -198,12 +200,167 @@ def searchByLikeNameType(self, name: str, typ: str, limit: int) -> Iterable[Dict
+        an artifact.
+        """
+        artifact: List[Dict[str, str]] = []
+        if isinstance(key, UUID):
+            artifact = list(self.get_artifact_by_uuid(key))
+        else:
+            # This is a hash.
+            artifact = list(self.get_artifact_by_hash(key))
+        return artifact[0]
+
+    def downloadFile(self, key: UUID, path: Path) -> None:
+        """Do nothing."""
+        pass
+
+    def searchByName(self, name: str, limit: int) -> Iterable[Dict[str, Any]]:
+        """Returns an iterable of all artifacts in the database that match
+        some name."""
+        raise Exception("Not implemented!")
+
+    def searchByType(self, typ: str, limit: int) -> Iterable[Dict[str, Any]]:
+        """Returns an iterable of all artifacts in the database that match
+        some type."""
+        raise Exception("Not implemented!")
+
+    def searchByNameType(self, name: str, typ: str, limit: int) -> Iterable[Dict[str, Any]]:
+        """Returns an iterable of all artifacts in the database that match
+        some name and type."""
+        raise Exception("Not implemented!")
+
+    def searchByLikeNameType(self, name: str, typ: str, limit: int) -> Iterable[Dict[str, Any]]:
+        """Returns an iterable of all artifacts in the database that match
+        some type and a regex name."""
+        raise Exception("Not implemented!")
+
+    def _load_from_file(self, json_file: Path) -> Tuple[Dict[str, Dict[str,str]], Dict[str, List[str]]]:
+        uuid_mapping: Dict[str, Dict[str,str]] = {}
+        hash_mapping: Dict[str, List[str]] = {}
+        if json_file.exists():
+            with open(json_file, 'r') as f:
+                j = json.load(f)
+                self._hash_uuid_map = j['hashes']
+                self._uuid_artifact_map = j['artifacts']
+        return uuid_mapping, hash_mapping
+    
+    def _save_to_file(self, json_file: Path) -> None:
+        content = {'artifacts': self._uuid_artifact_map,
+                   'hashes': self._hash_uuid_map}
+        with open(json_file, 'w') as f:
+            json.dump(content, f, indent=4, cls=ArtifactFileDB.ArtifactEncoder)
+    
+    def has_uuid(self, the_uuid: UUID) -> bool:
+        return str(the_uuid) in self._uuid_artifact_map
+    
+    def has_hash(self, the_hash: str) -> bool:
+        return the_hash in self._hash_uuid_map
+    
+    def get_artifact_by_uuid(self, the_uuid: UUID) -> Iterable[Dict[str,str]]:
+        uuid_str = str(the_uuid)
+        if not uuid_str in self._uuid_artifact_map:
+            return
+        yield self._uuid_artifact_map[uuid_str]
+
+    def get_artifact_by_hash(self, the_hash: str) -> Iterable[Dict[str,str]]:
+        if not the_hash in self._hash_uuid_map:
+            return
+        for the_uuid in self._hash_uuid_map[the_hash]:
+            yield self._uuid_artifact_map[the_uuid]
+    
+    def register_file(self, key: UUID, path: Path) -> None:
+        pass
```

Comment:
What is this function? Should be documented or removed

Reply:
I should be used by the upload() function. The blocker is that I don't see how to find the md5sum of it without having to check the md5sum again, which should have been done earlier, I think.

Reply:
ah, I could have used the UUID to find the hash

Reply:
ah, no need to do that